### PR TITLE
read-resource operation does not resolve expressions recursively code…

### PR DESCRIPTION
**Pull request by swetapatra**
… and test cases as part of WFCORE-4923

Thanks for submitting your Pull Request!

https://issues.redhat.com/browse/WFCORE-4923

### CODE REVIEW

**Comment by swetapatra** [comment]
@jmesnil @darranl kindly review this,the other branch is somehow corrupted even a Hard reset is not getting it to work,jenkins test cases keep failing on it

File: controller/src/test/java/org/jboss/as/controller/test/GlobalOperationsTestCase.java [hunk 1]
```diff
@@ -953,6 +954,19 @@ public void testReadResourceAttributesOnly() throws Exception {
...
+            try {   ModelNode operation = createOperation(READ_RESOURCE_OPERATION, "profile", "profileA");
+            System.setProperty("expr", "test-resolve-true");
+            operation.get(RECURSIVE).set(true);
+            operation.get(RESOLVE_EXPRESSIONS).set(true);
+            ModelNode result = executeForResult(operation);
+            assertTrue(result.toString(), result.hasDefined("subsystem", "subsystem2"));
+            checkRecursiveSubsystem2(result.get("subsystem", "subsystem2"), "test-resolve-true");
+        } finally {
+            System.clearProperty("expr");
+        }     }
```
Comment (jmesnil) [inline controller/src/test/java/org/jboss/as/controller/test/GlobalOperationsTestCase.java:968]: please fix the code formatting

**Review by jmesnil — CHANGES_REQUESTED** [review]

**Comment by swetapatra** [comment]
@jmesnil done with the changes squashed it into the same commit as well.

**Comment by swetapatra** [comment]
@jmesnil @darranl @bstansberry could you kindly review and let me know if any further changes are required?

**Comment by bstansberry** [comment]
Thanks, @swetapatra

**Pull request merged**